modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/mappings/NestedDDMFieldArrayUtil.java

```diff
@@ -43,14 +42,18 @@ public static Field createField(
 		return field;
 	}
 
-	public static Optional<Object> getFieldValue(
-		String name, Stream<Map<String, Object>> stream) {
+	public static Object getFieldValue(
+		String name, List<Map<String, Object>> maps) {
 
-		return stream.filter(
-			map -> name.equals(map.get("ddmFieldName"))
-		).map(
-			map -> map.get(map.get("ddmValueFieldName"))
-		).findAny();
+		for (Map<String, Object> map : maps) {
+			if (name.equals(map.get("ddmFieldName")) &&
+				(map.get(map.get("ddmValueFieldName")) != null)) {
+
+				return map;
```

Comment:
@julschong You should return the value of map.get(map.get("ddmValueFieldName"), not the map.

Please check if any test is using this logic as you passed ci test.

Tina.

Reply:
Sorry for the mix up...

Unit test did fail here
`com.liferay.portal.search.elasticsearch7.internal.sort.ElasticsearchSortFieldTranslatorTest > testFieldSortWithNestedSort FAILED
    org.junit.ComparisonFailure: {"from":0,"size":20,"query":{"bool":{"must":[{"term":{"entryClassName":{"value":"elasticsearchsortfieldtranslatortest.testfieldsortwithnestedsort"}}},{"term":{"companyId":{"value":"3730421925471104915"}}}]}},"explain":false,"_source":{"includes":["ddmFieldArray.*"],"excludes":[]},"stored_fields":"*","sort":[{"ddmFieldArray.ddmFieldValueKeyword":{"order":"asc","unmapped_type":"keyword","nested":{"path":"ddmFieldArray"}}}],"track_total_hits":2147483647} expected:<[[A, B, C]]> but was:<[[{ddmFieldName=nestedSortWithFieldSort, ddmValueFieldName=ddmFieldValueKeyword, ddmFieldValueKeyword=A}, {ddmFieldName=nestedSortWithFieldSort, ddmValueFieldName=ddmFieldValueKeyword, ddmFieldValueKeyword=B}, {ddmFieldName=nestedSortWithFieldSort, ddmValueFieldName=ddmFieldValueKeyword, ddmFieldValueKeyword=C}]]>`

will resend this